README.md

```diff
@@ -4,37 +4,680 @@ Sylvia is the old name meaning Spirit of The Wood.
+
+#[contract(error=ContractError)]
+impl MyContract<'_> {
+    #[msg(query)]
+    pub fn counter(&self, ctx: QueryCtx) -> StdResult<CounterResp> {
+        self
+            .counter
+            .load(ctx.deps.storage)
+            .map(|counter| CounterResp { counter })
+    }
+}
+```
+
+What you might notice - we can still use `StdResult` (so `StdError`) if we don't
+need `ContractError` in a particular function. What is important is that the returned
+result type has to implement `Into<ContractError>`, where `ContractError` is a contract
+error type - it will all be commonized in the generated dispatching function (so
+entry points have to return `ContractError` as its error variant).
+
+Messages equivalent to execution messages are generated. Let's create an entry point:
+
+```rust
+use sylvia::cw_std::{Deps, Binary};
+
+#[entry_point]
+fn query(deps: Deps, env: Env, msg: ContractQueryMsg) -> Result<Binary, ContractError> {
+    msg.dispatch(&CONTRACT, (deps, env))
+}
+```
+
+## Interfaces
+
+One of the fundamental ideas of Sylvia's framework is interfaces, allowing the
+grouping of messages into their semantical groups. Let's define a Sylvia interface:
+
+```rust
+pub mod group {
+    use super::*;
+    use sylvia::interface;
+    use sylvia::types::ExecCtx;
+    use sylvia::cw_std::StdError;
+
+    #[cw_serde]
+    pub struct IsMemberResp {
+        pub is_member: bool,
+    }
+
+    #[interface]
+    pub trait Group {
+        type Error: From<StdError>;
+
+        #[msg(exec)]
+        fn add_member(&self, ctx: ExecCtx, member: String) -> Result<Response, Self::Error>;
+
+        #[msg(query)]
+        fn is_member(&self, ctx: QueryCtx, member: String) -> Result<IsMemberResp, Self::Error>;
+    }
+}
+```
+
+Then we need to implement the trait on the contract type:
+
+```rust
+use sylvia::cw_std::{Empty, Addr};
+use cw_storage_plus::{Map, Item};
+
+pub struct MyContract<'a> {
+    counter: Item<'a, u64>,
+    // New field added - remember to initialize it in `new`
+    members: Map<'a, &'a Addr, Empty>,
+}
+
+#[contract]
+#[messages(group as Group)]
+impl group::Group for MyContract<'_> {
+    type Error = ContractError;
+
+    #[msg(exec)]
+    fn add_member(&self, ctx: ExecCtx, member: String) -> Result<Response, ContractError> {
+        let member = ctx.deps.api.addr_validate(member)?;
+        self.members.save(ctx.deps.storage, &member, &Empty {})?;
+        Ok(Response::new())
+    }
+
+    #[msg(query)]
+    fn is_member(&self, ctx: QueryCtx, member: String) -> Result<group::IsMemberResp, ContractError> {
+        let is_member = self.members.has(ctx.deps.storage, &Addr::unchecked(&member));
+        let resp = group::IsMemberResp {
+            is_member,
+        };
+
+        Ok(resp)
+    }
+}
+
+#[contract]
+#[messages(group as Group)]
+impl MyContract<'_> {
+    // Nothing changed here
+}
+```
+
+Here are a couple of things to talk about.
+
+First, note that I defined the interface trait in its separate module with a name
```

Comment:
Please explain this a bit more in detail with examples:

`mod group` -> `trait Group`
`mod cross_staking` -> `trait CrossStaking`